main.tf

```diff
@@ -5,12 +5,18 @@ data "aws_vpc" "vpc" {
 data "aws_region" "current" {
 }
 
-data "aws_ami_ids" "ami" {
-  owners = ["amazon"]
+data "aws_ami" "amazon-linux-2" {
+  owners      = ["amazon"]
+  most_recent = true
```

Comment:
```suggestion
```
This isn't an valid attribute, https://www.terraform.io/docs/providers/aws/d/ami_ids.html#sort_ascending is probably what you were thinking of but that defaults to descending which is what we want.